airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -934,13 +934,22 @@ def var_print(var_name):
         else:
             return "Format exception for {var}: ".format(var=var_name)
 
-    cmpt = table_input.split(':')
+    if table_input.count('.') + table_input.count(':') > 3:
```

Comment:
Actually the last one is not valid AFAIK, a project_id can only have a single ':' character in it denoting the org, so for:
some:other:proj:datset.table

In this example the project id is: "some:other:proj" which is invalid, and would be caught by this validation.
The rest of your examples are valid I believe and should work with this patch.